microsoft/surface/repos.nix

```diff
@@ -1,8 +1,29 @@
-{ lib, pkgs, fetchgit }:
+{ lib, pkgs, fetchFromGitHub, fetchurl }:
 {
-  linux-surface = fetchgit {
-    url="https://github.com/linux-surface/linux-surface.git";
+  linux-surface = fetchFromGitHub {
+    owner="linux-surface";
+    repo="linux-surface";
     rev="f8fab978a480a4ed57e9ebb6928683b2e443c1c5";
     sha256="0zwybprwjckpapxm6gxzh6hwdd1w91g5sjxn6z52zlvvjpkmw959";
```

Comment:
Please check what if the has is correct here. It should change if you go from `fetchgit` to `fetchFromGitHub`. You can check by changing one character in the hash and nix still producing the same hash.

Reply:
I just rechecked it:

```
% nix-prefetch-github linux-surface linux-surface --rev f8fab978a480a4ed57e9ebb6928683b2e443c1c5
{
    "owner": "linux-surface",
    "repo": "linux-surface",
    "rev": "f8fab978a480a4ed57e9ebb6928683b2e443c1c5",
    "sha256": "0zwybprwjckpapxm6gxzh6hwdd1w91g5sjxn6z52zlvvjpkmw959",
    "fetchSubmodules": true
}
```

Reply:
Yes, but the difference is if you use `fetchSubmodules` than it is actually using `fetchgit`.

Reply:
But in this case it seems fine.

Reply:
> Yes, but the difference is if you use `fetchSubmodules` than it is actually using `fetchgit`.

Which I'm guessing might be part of why it worked?
AFAICT, the default is `false` for `fetchForGitHub`?

> But in this case it seems fine.
I have tested rebuilding+installing this twice on a Surface Go — as that's kind-of my whole reason for caring about this!